src/os/xdg.zig

```diff
@@ -30,6 +32,17 @@ pub fn config(alloc: Allocator, opts: Options) ![]u8 {
 
 /// Get the XDG cache directory. The returned value is allocated.
 pub fn cache(alloc: Allocator, opts: Options) ![]u8 {
+    if (posix.getenv("XDG_CACHE_HOME")) |env| {
+        return try std.fs.path.join(alloc, &[_][]const u8{
+            env,
+            opts.subdir orelse "",
+        });
+    }
+
+    if (builtin.os.tag == .macos) {
+        return try macos.cacheDir(alloc, opts.subdir orelse "");
+    }
+
     return try dir(alloc, opts, .{
```

Comment:
This feels redundant given the work that `dir(...)` does below.

A few thoughts:

- Maybe we should instead teach `InternalOptions` about `NSSearchPathDirectory`?
- Or maybe this file should remain XDG-specific, and we add our macOS-aware logic to a higher-level call?

Reply:
The XDG/macOS architecture suggestions are interesting, but I think they deserve their own discussion and PR to avoid scope creep in this bugfix. What do you think?

Reply:
I generally agree and don't want to turn this change into something bigger. That's something one of us can look at later.

I'm more concerned about making sure any new code that we drop into this existing path behaves nicely in the interim.